sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/ScalaReflection.scala

```diff
@@ -94,7 +94,7 @@ object ScalaReflection extends ScalaReflection {
    * JVM form instead of the Scala Array that handles auto boxing.
    */
   private def arrayClassFor(tpe: `Type`): ObjectType = {
```

Comment:
arrayClassFor is called at many place. The typical calling pattern looks like:

            val TypeRef(_, _, Seq(elementType)) = tpe
            arrayClassFor(elementType)

So instead of dealiasing when calling, we dealiase it here.